Login OIDC (#173)
epinio/docs: docs/references/authentication_oidc.md

---

> sidebar_label: "Authentication"

Reviewer: ```suggestion
sidebar_label: "OIDC authentication"
```

title: ""
---

> # Authentication

Reviewer: ```suggestion
# OIDC authentication
```

> Since version **1.3.0** Epinio has integrated [Dex](https://dexidp.io/) as identity provider, adding supports for external OIDC providers.

Reviewer: ```suggestion
Since version **1.3.0**, Epinio has integrated [Dex](https://dexidp.io/) as an identity provider which adds the support for external OIDC providers.
```

To authenticate through it you can use the login command with the `--oidc` flag. This will open a web page where you can authenticate with the configured providers.

```bash
epinio login --oidc https://epinio.mydomain.com
```

If you are using the `epinio` cli without a desktop environment you can provide the `--prompt` flag. This will give you the url where you can authenticate, and it will wait for the authorization code that you will get at the end of the authentication.

```bash
epinio login --oidc --prompt https://epinio.172.21.0.4.omg.howdoi.website
```

> By default only the local connector is setup, with a two users (`admin@epinio.io` and `epinio@epinio.io`).
> To add more connectors you can edit the Dex config file, deployed with the `dex-config` secret in the `epinio` namespace.

Reviewer: ```suggestion
By default, only the local connector is setup with two users (`admin@epinio.io` and `epinio@epinio.io`).
To add more connectors, you can edit the Dex config file deployed with the `dex-config` secret in the `epinio` namespace.
```

> After a successful login a new Epinio user will be created, with the username matching the email used to login, and returned by the provider.

Reviewer: ```suggestion
After a successful login, a new Epinio user will be created with the username matching the email used to login and returned by the provider.
```
